### ISSUE TITLE
HttpUtility.ParseQueryString doesn't roundtrip when there are keys that only differ by case

### ISSUE THREAD

**Issue by cmeeren** [dotnet/corefx#41732]
Using .NET Core 3.0, `HttpUtility.ParseQueryString` doesn't rountrip when there are keys in the query string that only differ by case.

```f#
let input = "type=foo&TYPE=bar"
let values = System.Web.HttpUtility.ParseQueryString(input)
let output = values.ToString()
Assert.Equal(input, output)  // not equal; is "type=foo&type=bar"
```

AFAIK [query strings are case sensitive](https://stackoverflow.com/a/24700171/2978652).

Possibly related: https://github.com/dotnet/corefx/issues/32575

**Comment by scalablecory** [comment]
@cmeeren can you check if this behavior existed in .NET Core 2.2 and Framework 4.8?

**Comment by cmeeren** [comment]
Code:


```f#
[<EntryPoint>]
let main argv =
  let input = "type=foo&TYPE=bar"
  let values = System.Web.HttpUtility.ParseQueryString(input)
  printfn "input: %s" input
  printfn "output: %s" <| values.ToString()
  0
```

`netcoreapp3.0` as described:

```
input: type=foo&TYPE=bar
output: type=foo&type=bar
```

`netcoreapp2.2` suffers from the problem in #32575:

```
input: type=foo&TYPE=bar
output: type=foo%2cbar
```

`net48`, `net472`, `net461` (all with `<Reference Include="System.Web" />`) are same as `netcoreapp3.0`:

```
input: type=foo&TYPE=bar
output: type=foo&type=bar
```

**Comment by scalablecory** [comment]
Thanks.

This does look like a bug, but it would be a breaking change for anyone expecting case insensitivity when querying keys.

A new API on `Uri` might be better. The feature would be more discoverable that way as well. What do you think of that?

**Comment by cmeeren** [comment]
I think that sounds like a wonderful idea.

In general, my main underlying pain point here is manipulating a URI, e.g. by adding path segments (appending/stripping slashes as needed) without modifying anything else (notably the query/fragment), adding (appending/replacing) query parameters, etc. I did this using `UriBuilder` (which had other problems as described in #41679) and `HttpUtility.ParseQueryString` (which has the bug described here), so at the moment I have hacked something together to manually perform these tasks for the kinds of URLs I expect to come across. Not an optimal or general solution.

Perhaps the `Uri` API could be extended with methods to append path segments, append/remove/replace query parameters, etc.?

In any case, just being able to parse query strings correctly using a new `Uri` API would help.
